Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/IConceptTypesService.java

```diff
@@ -0,0 +1,14 @@
+package edu.asu.conceptpower.app.core;
+
+import edu.asu.conceptpower.app.core.impl.ConceptTypesService.ConceptTypes;
+
+public interface IConceptTypesService {
+
+    /**
+     * This method returns the concept type based on the id parameter.
+     * 
+     * @param id
+     * @return ConceptTypes
+     */
+    public ConceptTypes getConceptTypeByConceptId(String id);
```

Comment:
The return type is defined in the class implementing this interface => Circular dependency (class implements interface, but interface needs type defined in class) == bad.